modules/new_serverpod_auth/serverpod_auth_profile/serverpod_auth_profile_server/lib/src/business/user_profiles.dart

```diff
@@ -0,0 +1,155 @@
+import 'package:serverpod/serverpod.dart';
+import 'package:serverpod_auth_profile_server/serverpod_auth_profile_server.dart';
+
+/// Business logic for handling user profiles
+abstract final class UserProfiles {
+  /// Creates a new user and stores it in the database.
+  static Future<UserProfile> createUserProfile(
+    final Session session,
+    UserProfile userProfile,
+  ) async {
+    userProfile = (await UserProfileConfig.current.onBeforeUserProfileCreated
+            ?.call(session, userProfile)) ??
+        userProfile;
+
+    userProfile = userProfile.id == null
+        ? await UserProfile.db.insertRow(session, userProfile)
+        : await UserProfile.db.updateRow(session, userProfile);
+
+    await UserProfileConfig.current.onAfterUserProfileCreated?.call(
+      session,
+      userProfile,
+    );
+
+    return userProfile;
+  }
+
+  /// Finds a user profile by its email address.
+  ///
+  /// Returns `null` if no user profile is found.
+  static Future<UserProfile?> maybeFindUserProfileByEmail(
+    final Session session,
+    final String email,
+  ) async {
+    return await UserProfile.db.findFirstRow(
+      session,
+      where: (final t) => t.email.ilike(email),
```

Comment:
This seems like it could get too query-heavy.

So we should either:
- Always lower-cases the email on account creation, and then lower-case all queries
- Store both the original input and lowercase (normalized?) version (Would be interesting, if there is a popular library that would also handle `@googlemail` vs `@gmail` etc.)

Reply:
lower-case full-string matching it is for now :)